<a href="https://colab.research.google.com/github/Playmen998/Probability_problems_bootstrap/blob/main/Probability_problems_bootstrap.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Введение

В данном ноутбуке решены задачи на теорию вероятности методом bootstrap. Задачи брались из ЕГЭ на сайте https://math-ege.sdamgia.ru/test?theme=185 и из других источниках

# Задача 1

Какова вероятность того, что случайно выбранный телефонный номер оканчивается двумя чётными цифрами?  
https://math-ege.sdamgia.ru/problem?id=509352

In [2]:
import pandas as pd
import numpy as np

In [3]:
from tqdm.notebook import tqdm

In [90]:
from random import shuffle

In [4]:
import matplotlib.pyplot as plt
plt.style.use('dark_background')

In [ ]:
number = [0,1,2,3,4,5,6,7,8,9]

In [ ]:
nums = pd.Series(number).sample(2,replace = True).values

In [ ]:
result = []
for _ in range(100000):
  nums = pd.Series(number).sample(2,replace = True).values
  if (nums[0] % 2 == 0) and (nums[1] % 2 == 0):
    result.append(1)
  else:
    result.append(0)

In [ ]:
np.mean(result)

0.25087

Вероятность того, что телефонный номер оканчивается двумя чётными цифрами = 0.25

# Задача 2

Если шахматист А. играет белыми фигурами, то он выигрывает у шахматиста Б. с вероятностью 0,52. Если А. играет черными, то А. выигрывает у Б. с вероятностью 0,3. Шахматисты А. и Б. играют две партии, причём во второй партии меняют цвет фигур. Найдите вероятность того, что А. выиграет оба раза.  
https://math-ege.sdamgia.ru/problem?id=319355

In [ ]:
result = []
for _ in range(1000000):
  one_game = np.random.binomial(1,0.52)
  two_game = np.random.binomial(1,0.3)
  result.append(one_game+two_game)

In [ ]:
pd.Series(result).value_counts(normalize=True)

1    0.507993
0    0.335857
2    0.156150
dtype: float64

Вероятность выйграша шахматиста А в двух партиях равна 0.156

# Задача 3

Вероятность того, что батарейка бракованная, равна 0,06. Покупатель в магазине выбирает случайную упаковку, в которой две таких батарейки. Найдите вероятность того, что обе батарейки окажутся исправными.  
https://math-ege.sdamgia.ru/problem?id=320210

In [ ]:
result = []
for _ in tqdm(range(1000000)):
  one_reject = np.random.binomial(1,0.06)
  two_reject = np.random.binomial(1,0.06)
  result.append(one_reject+two_reject)

  0%|          | 0/1000000 [00:00<?, ?it/s]

In [ ]:
pd.Series(result).value_counts(normalize=True)

0    0.883554
1    0.112862
2    0.003584
dtype: float64

Вероятность того, что обе батарейки исправные = 0.883  
Для того чтобы получить более точный результат необходимо увеличить выборку

# Задача 4

Биатлонист пять раз стреляет по мишеням. Вероятность попадания в мишень при одном выстреле равна 0,8. Найдите вероятность того, что биатлонист первые три раза попал в мишени, а последние два промахнулся. Результат округлите до сотых.  
https://math-ege.sdamgia.ru/problem?id=320173 

In [ ]:
result = []
for _ in tqdm(range(500000)):
  result.append(str(np.random.binomial(1,0.8,size=5)))

  0%|          | 0/500000 [00:00<?, ?it/s]

In [ ]:
pd.Series(result).value_counts(normalize=True)

[1 1 1 1 1]    0.327494
[0 1 1 1 1]    0.082316
[1 1 1 1 0]    0.082122
[1 0 1 1 1]    0.081992
[1 1 1 0 1]    0.081814
[1 1 0 1 1]    0.081684
[1 1 1 0 0]    0.020560
[1 0 1 1 0]    0.020558
[0 0 1 1 1]    0.020452
[1 1 0 1 0]    0.020408
[1 1 0 0 1]    0.020406
[1 0 1 0 1]    0.020394
[1 0 0 1 1]    0.020376
[0 1 1 0 1]    0.020366
[0 1 0 1 1]    0.020280
[0 1 1 1 0]    0.020106
[1 0 0 0 1]    0.005238
[0 0 0 1 1]    0.005226
[1 1 0 0 0]    0.005190
[0 0 1 0 1]    0.005184
[0 1 1 0 0]    0.005166
[1 0 0 1 0]    0.005150
[0 1 0 0 1]    0.005132
[1 0 1 0 0]    0.005128
[0 1 0 1 0]    0.005082
[0 0 1 1 0]    0.005022
[0 0 1 0 0]    0.001418
[1 0 0 0 0]    0.001386
[0 0 0 1 0]    0.001348
[0 0 0 0 1]    0.001340
[0 1 0 0 0]    0.001328
[0 0 0 0 0]    0.000334
dtype: float64

In [ ]:
pd.Series(result).value_counts(normalize=True)['[1 1 1 0 0]']

0.02056

Вероятность того, что биатлонист попадет 3 разя подряд, а потом 2 раза промахнется по мишени, составляет 0.2 

# Задача 5

В магазине три продавца. Каждый из них занят с клиентом с вероятностью 0,3. Найдите вероятность того, что в случайный момент времени все три продавца заняты одновременно (считайте, что клиенты заходят независимо друг от друга).  
https://math-ege.sdamgia.ru/problem?id=320201

In [ ]:
result = []
for _ in range(10000):
  result.append(str(np.random.binomial(1,0.3,size=3)))

In [ ]:
pd.Series(result).value_counts(normalize=True)

[0 0 0]    0.3409
[0 0 1]    0.1487
[0 1 0]    0.1468
[1 0 0]    0.1423
[0 1 1]    0.0667
[1 0 1]    0.0642
[1 1 0]    0.0633
[1 1 1]    0.0271
dtype: float64

Вероятность, того что сразу 3 продавца будут заняты, равна 0.027

# Задача 6

Ковбой Джон попадает в муху на стене с вероятностью 0,9, если стреляет из пристрелянного револьвера. Если Джон стреляет из непристрелянного револьвера, то он попадает в муху с вероятностью 0,2. На столе лежит 10 револьверов, из них только 4 пристрелянные. Ковбой Джон видит на стене муху, наудачу хватает первый попавшийся револьвер и стреляет в муху. Найдите вероятность того, что Джон промахнётся.  
https://math-ege.sdamgia.ru/problem?id=320180

In [ ]:
result = []
for _ in range(1000000):  
  sum_gun = [np.random.binomial(1,0.2) for _ in range(6)] + [np.random.binomial(1,0.9) for _ in range(4)]
  shuffle(sum_gun)
  result.append(sum_gun[0])

In [ ]:
pd.Series(result).value_counts(normalize=True)

0    520020
1    479980
dtype: int64

Вероятность **не** попадание ковбоем в муху равна 0.52

# Задача 7

Две фабрики выпускают одинаковые стекла для автомобильных фар. Первая фабрика выпускает 45% этих стекол, вторая — 55%. Первая фабрика выпускает 3% бракованных стекол, а вторая — 1%. Найдите вероятность того, что случайно купленное в магазине стекло окажется бракованным.  
https://math-ege.sdamgia.ru/problem?id=319353

In [ ]:
result = []
for _ in tqdm(range(1000000)): 
  sum = np.hstack([np.random.binomial(1,0.97,45), np.random.binomial(1,0.99,55)])
  shuffle(sum) 
  result.append(sum[0])

  0%|          | 0/1000000 [00:00<?, ?it/s]

In [ ]:
pd.Series(result).value_counts(normalize=True)

1    0.98114
0    0.01886
dtype: float64

Вероятность купить бракованное стекло составляет 0.019

# Задача 8

Всем пациентам с подозрением на гепатит делают анализ крови. Если анализ выявляет гепатит, то результат анализа называется положительным. У больных гепатитом пациентов анализ даёт положительный результат с вероятностью 0,9. Если пациент не болен гепатитом, то анализ может дать ложный положительный результат с вероятностью 0,01. Известно, что 5% пациентов, поступающих с подозрением на гепатит, действительно больны гепатитом. Найдите вероятность того, что результат анализа у пациента, поступившего в клинику с подозрением на гепатит, будет положительным.  
https://math-ege.sdamgia.ru/problem?id=320207

In [ ]:
def sick(pat):
  if pat:
    return np.random.binomial(1,0.9) # точно больной
  else:
    return np.random.binomial(1,0.01)

In [ ]:
result = []
for _ in range(1000000):
  pat = np.random.binomial(1,0.05)
  result.append(sick(pat))

In [ ]:
pd.Series(result).value_counts(normalize=True)

0    0.945428
1    0.054572
dtype: float64

Найдите вероятность того, что результат анализа у пациента, поступившего в клинику с подозрением на гепатит, будет положительным равна 0.0545

# Задача 9

В кармане у Пети было 2 монеты по 5 рублей и 4 монеты по 10 рублей. Петя, не глядя, переложил какие-то 3 монеты в другой карман. Найдите вероятность того, что пятирублевые монеты лежат теперь в разных карманах.  
https://math-ege.sdamgia.ru/problem?id=500998

In [ ]:
result = []
for _ in range(100000):
  pocket_1 = [5,5,10,10,10,10]
  shuffle(pocket_1) 
  pocket_2 = pocket_1[:3]
  del pocket_1[:3]
  if (5 in pocket_1) and (5 in pocket_2):
    result.append(1)
  else:
    result.append(0)

In [ ]:
pd.Series(result).value_counts(normalize=True)

1    0.59999
0    0.40001
dtype: float64

Вероятность того, что пятирублевые монеты лежат теперь в разных карманах равна 0.6

# Задача 10

Стрелок стреляет по мишени один раз. В случае промаха стрелок делает второй выстрел по той же мишени. Вероятность попасть в мишень при одном выстреле равна 0,7. Найдите вероятность того, что мишень будет поражена (либо первым, либо вторым выстрелом).  
https://math-ege.sdamgia.ru/problem?id=501061

In [ ]:
result = []
for _ in tqdm(range(1000000)):
  result.append(str(np.random.binomial(1,0.7,size=2)))

  0%|          | 0/1000000 [00:00<?, ?it/s]

In [ ]:
df = pd.DataFrame(result).value_counts(normalize=True).sort_values()
df = df.to_frame()
df.columns = ['Event']
df['Not_event'] = 1 - df['Event']

In [ ]:
df

Event  Not_event
0                        
[0 0]  0.09018    0.90982
[1 0]  0.20996    0.79004
[0 1]  0.21098    0.78902
[1 1]  0.48888    0.51112

Вероятность того, что стрелок не попадет 2 выстрела подряд равна 0.09, получается противоположное событие (когда он попадет за 2 выстрела) равно 0.91

# Задача 11

Игральный кубик бросают дважды. Известно, что в сумме выпало 8 очков. Найдите вероятность того, что во второй раз выпало 3 очка.  
https://math-ege.sdamgia.ru/problem?id=508754

In [ ]:
dice = pd.Series([1,2,3,4,5,6])

In [ ]:
def d():
  return dice.sample(1,replace=True).values[0]

In [ ]:
result = []
for _ in tqdm(range(100000)):
  one = d()
  two = d()
  if (one + two) == 8:
    result.append(str([one,two]))

  0%|          | 0/100000 [00:00<?, ?it/s]

In [ ]:
pd.Series(result).value_counts(normalize=True)

[3, 5]    0.204870
[4, 4]    0.201228
[2, 6]    0.199729
[6, 2]    0.197658
[5, 3]    0.196515
dtype: float64

Вероятность того, что во второй раз выпало 3 очка равна 0.2

# Задача 12

Игральную кость бросили один или несколько раз. Оказалось, что сумма всех выпавших очков равна 4. Какова вероятность того, что был сделан один бросок? Ответ округлите до сотых.  
https://math-ege.sdamgia.ru/problem?id=508792

In [ ]:
dice = pd.Series([1,2,3,4,5,6])

In [ ]:
def d():
  return dice.sample(1,replace=True).values[0]

In [ ]:
result = []
for _ in tqdm(range(1000000)):  
  sum = 0
  shot = 0
  while sum < 4:
      sum += d()
      shot += 1
  if sum == 4:
      result.append(shot)

  0%|          | 0/1000000 [00:00<?, ?it/s]

In [ ]:
pd.Series(result).value_counts(normalize=True)

1    0.629285
2    0.315481
3    0.052356
4    0.002877
dtype: float64

Вероятность того, что был сделан один бросок 0.629 (ответ может не совпадать до сотых значений, для этого нужно увеличить выборку)

# Задача 13

Первый игральный кубик обычный, а на гранях второго кубика нет чётных чисел, а нечётные числа 1, 3 и 5 встречаются по два раза. В остальном кубики одинаковые. Один случайно выбранный кубик бросают два раза. Известно, что в каком-то порядке выпали 3 и 5 очков. Какова вероятность того, что бросали второй кубик?  
https://math-ege.sdamgia.ru/problem?id=508887

In [ ]:
dice_1 = pd.Series([1,2,3,4,5,6])

In [ ]:
def d_1():
  return dice_1.sample(1,replace=True).values[0]

In [ ]:
dice_2 = pd.Series([1,1,3,3,5,5])

In [ ]:
def d_2():
  return dice_2.sample(1,replace=True).values[0]

In [ ]:
result = []
for _ in tqdm(range(100000)):
  sum = []
  if np.random.binomial(1,0.5):
    sum.append(d_1())
    sum.append(d_1())
    if (sum[0] == 3 and sum[1] == 5) or (sum[0] == 5 and sum[1] == 3):
      result.append(1)
  else:
    sum.append(d_2())
    sum.append(d_2())
    if (sum[0] == 3 and sum[1] == 5) or (sum[0] == 5 and sum[1] == 3):
      result.append(2)

In [ ]:
pd.Series(result).value_counts(normalize=True)

2    0.795455
1    0.204545
dtype: float64

Вероятность выпадения 2 кубика = 0.8

# Задача 14

Маша коллекционирует принцесс из Киндер-сюрпризов. Всего в коллекции 10 разных принцесс, и они равномерно распределены, то есть в каждом очередном Киндер-сюрпризе может с равными вероятностями оказаться любая из 10 принцесс. У Маши уже есть две разные принцессы из коллекции. Какова вероятность того, что для получения следующей принцессы Маше придётся купить ещё 2 или 3 шоколадных яйца?  
https://math-ege.sdamgia.ru/problem?id=509078

In [70]:
doll = pd.Series([1,2,3,4,5,6,7,8,9,10])

In [71]:
def egg():
  return doll.sample(1,replace=True).values[0]

In [72]:
have = [1,3]

In [73]:
result = []
for _ in range(100000):
  '2 покупки'
  sum = []
  for _ in range(2):
    if egg() not in have:
      sum.append(1)
    else:
      sum.append(0)
  result.append(str(sum))

In [74]:
pd.Series(result).value_counts(normalize=True)

[1, 1]    0.63982
[0, 1]    0.16066
[1, 0]    0.16037
[0, 0]    0.03915
dtype: float64

0.16 вероятность получить новую куклу при 2 покупках

In [75]:
result = []
for _ in range(100000):
  '3 покупки'
  sum = []
  for _ in range(3):
    if egg() not in have:
      sum.append(1)
    else:
      sum.append(0)
  result.append(str(sum))

In [76]:
pd.Series(result).value_counts(normalize=True)

[1, 1, 1]    0.51196
[1, 0, 1]    0.12805
[0, 1, 1]    0.12744
[1, 1, 0]    0.12717
[1, 0, 0]    0.03296
[0, 1, 0]    0.03222
[0, 0, 1]    0.03215
[0, 0, 0]    0.00805
dtype: float64

0.032 вероятность получить новую куклу при 2 покупках

Видим для 2 покупок вероятность получить новую куклу 0.16 Для 3 покупок вероятность получить новую куклу 0.032, значит вероятность первого или второго события равны 0.16 + 0.032 =  0,192

# Задача 15

В коробке 8 синих, 6 красных и 11 зелёных фломастеров. Случайным образом выбирают два фломастера. Какова вероятность того, что окажутся выбраны один синий и один красный фломастер?  
https://math-ege.sdamgia.ru/problem?id=621772

In [93]:
box = pd.Series([1,1,1,1,1,1,1,1,2,2,2,2,2,2,3,3,3,3,3,3,3,3,3,3,3])
# 1 - синие
# 2 - красные
# 3 - зеленые

In [94]:
shuffle(box)

In [112]:
result = []
for _ in range(10000):
  result.append(str(box.sample(2).values))

In [113]:
pd.Series(result).value_counts(normalize=True)

[3 3]    0.1743
[3 1]    0.1556
[1 3]    0.1420
[2 3]    0.1137
[3 2]    0.1074
[1 1]    0.0953
[2 1]    0.0816
[1 2]    0.0810
[2 2]    0.0491
dtype: float64

Вероятность того, что первый фламастер красный, а второй синий [2 1], равна 0.08 и вероятность, что первый синий, а второй красный [1 2], равна 0.08. Получается вероятность первого или второго события равна 0.08 + 0.08 = 0.16 

# Задача 16

Стрелок стреляет по 4 одинаковым мишеням по одному разу, вероятность промаха 0,2, найдите вероятность что он попадёт в первую мишень, а в 3 оставшиеся промахнется.  
https://math-ege.sdamgia.ru/problem?id=630095

In [124]:
result = []
for _ in tqdm(range(1000000)):
  result.append(str(np.random.binomial(1,0.8,size=4)))

  0%|          | 0/1000000 [00:00<?, ?it/s]

In [125]:
pd.Series(result).value_counts(normalize=True)

[1 1 1 1]    0.409503
[1 1 0 1]    0.102781
[1 0 1 1]    0.102662
[0 1 1 1]    0.102354
[1 1 1 0]    0.101980
[0 0 1 1]    0.025964
[0 1 1 0]    0.025771
[1 0 1 0]    0.025617
[1 0 0 1]    0.025467
[1 1 0 0]    0.025445
[0 1 0 1]    0.025351
[0 0 0 1]    0.006470
[0 1 0 0]    0.006401
[1 0 0 0]    0.006391
[0 0 1 0]    0.006266
[0 0 0 0]    0.001577
dtype: float64

Вероятность что он попадёт в первую мишень, а в 3 оставшиеся промахнется, равна 0.0064

# Задача 17

При артиллерийской стрельбе автоматическая система делает выстрел по цели. Если цель не уничтожена, то система делает повторный выстрел. Выстрелы повторяются до тех пор, пока цель не будет уничтожена. Вероятность уничтожения некоторой цели при первом выстреле равна 0,4, а при каждом последующем — 0,6. Сколько выстрелов потребуется для того, чтобы вероятность уничтожения цели была не менее 0,98?  
https://math-ege.sdamgia.ru/problem?id=320187

**1 выстрел**

In [132]:
result = []
for _ in tqdm(range(100000)):
  if np.random.binomial(1,0.4):
    result.append(1)
  else:
    result.append(0)

  0%|          | 0/100000 [00:00<?, ?it/s]

In [133]:
pd.Series(result).value_counts(normalize=True)

0    0.5986
1    0.4014
dtype: float64

**2 выстрел**

In [134]:
result = []
for _ in tqdm(range(100000)):
  if np.random.binomial(1,0.4):
    result.append(1)
  else:
    if np.random.binomial(1,0.6):
      result.append(1)
    else:
      result.append(0)

  0%|          | 0/100000 [00:00<?, ?it/s]

In [135]:
pd.Series(result).value_counts(normalize=True)

1    0.76207
0    0.23793
dtype: float64

**3 выстрел**

In [142]:
result = []
for _ in tqdm(range(100000)):
  if np.random.binomial(1,0.4):
    result.append(1)
  else:
      if np.random.binomial(1,0.6):
        result.append(1)
      else:
        if np.random.binomial(1,0.6):
          result.append(1)
        else:
          result.append(0)

  0%|          | 0/100000 [00:00<?, ?it/s]

In [143]:
pd.Series(result).value_counts(normalize=True)

1    0.90426
0    0.09574
dtype: float64

**4 выстрел**

In [144]:
result = []
for _ in tqdm(range(100000)):
  if np.random.binomial(1,0.4):
    result.append(1)
  else:
      if np.random.binomial(1,0.6):
        result.append(1)
      else:
        if np.random.binomial(1,0.6):
          result.append(1)
        else:
          if np.random.binomial(1,0.6):
            result.append(1)
          else:
            result.append(0)

  0%|          | 0/100000 [00:00<?, ?it/s]

In [145]:
pd.Series(result).value_counts(normalize=True)

1    0.96203
0    0.03797
dtype: float64

**5 выстрел**

In [146]:
result = []
for _ in tqdm(range(100000)):
  if np.random.binomial(1,0.4):
    result.append(1)
  else:
      if np.random.binomial(1,0.6):
        result.append(1)
      else:
        if np.random.binomial(1,0.6):
          result.append(1)
        else:
          if np.random.binomial(1,0.6):
            result.append(1)
          else:
            if np.random.binomial(1,0.6):
              result.append(1)
            else:
              result.append(0)


  0%|          | 0/100000 [00:00<?, ?it/s]

In [147]:
pd.Series(result).value_counts(normalize=True)

1    0.98514
0    0.01486
dtype: float64

При 5 выстреле вероятность больше 0.98, а именно 0.98514

# Задача 18

Пародокс дня рождения. Определить вероятность того, что в группе, состоящей из 23 человек, у двух людей будет совпадение дней рождения (число и месяц).  
https://scienceparadoxes.fandom.com/ru/wiki/%D0%9F%D0%B0%D1%80%D0%B0%D0%B4%D0%BE%D0%BA%D1%81_%D0%B4%D0%BD%D0%B5%D0%B9_%D1%80%D0%BE%D0%B6%D0%B4%D0%B5%D0%BD%D0%B8%D1%8F

In [ ]:
bd = pd.Series(range(365)) # делаем все дни в году

In [153]:
daf = pd.Series([bd.sample(23,replace = True).duplicated().max() for i in range(100000)])

In [154]:
daf.value_counts(normalize=True)

True     0.50976
False    0.49024
dtype: float64

Вероятность того, что в группе, состоящей из 23 человек, у двух людей будет совпадение дней рождения, равно 0.5

# Задача 19

Экзамен проходит по следующей схеме: если некоторый билет уже был вытянут, то после ответа экзаменатор откладывает его в сторону. Студент выучил 20 билетов из 30. Когда ему выгоднее идти, первым или вторым, чтобы вероятность вынуть выученный билет была больше?

**Когда студент идет первым**

In [157]:
n = 1000000
student = list(range(1,21)) # выученные билеты
tickets = list(range(1,31)) # всего билетов
result = []

for _ in range(n):
  shuffle(tickets)
  result.append(tickets[0] in student)

In [158]:
np.mean(result)

0.665996

**Когда студент идет вторым**

In [159]:
n = 1000000
student = list(range(1,21)) # выученные билеты
tickets = list(range(1,31)) # всего билетов
result = []

for _ in range(n):
  shuffle(tickets)
  result.append(tickets[1] in student)

In [160]:
np.mean(result)

0.666341

**Когда студент идет десятым**

In [161]:
n = 1000000
student = list(range(1,21)) # выученные билеты
tickets = list(range(1,31)) # всего билетов
result = []

for _ in range(n):
  shuffle(tickets)
  result.append(tickets[9] in student)

In [162]:
np.mean(result)

0.66664

Разницы в вероятности не будет, если студент идет отвечать в любом порядке

# Задача 20

Ночью произошло ДТП с участием такси. В городе работают две компании такси — «Синие» и «Зелёные». «Зелёным» принадлежит 85% такси, «Синим» 15%. Свидетель аварии утверждает, что такси принадлежало «Синим». Следственный эксперимент показал, что ночью свидетель верно определяет цвет такси в 80%. Какова вероятность того, что такси действительно принадлежало «Синим»?  
https://pikabu.ru/story/zadachi_na_kotoryie_pravilno_otvetit_ne_kazhdyiy_5580394

In [163]:
def witness(taxi):
  if np.random.binomial(1,0.8): # если 1 то это правда и проходит if, иначе возварщает ошибку в определении такси
    return taxi
  return abs(taxi-1)

In [164]:
n = 10000
result = []
for _ in range(n):
  taxi = np.random.binomial(1,0.15) # фактическое такси
  # 1 - синее такси
  witness_answer = witness(taxi) # правильно ли определил свидетель такси да/нет
  result.append((taxi,witness_answer))

In [165]:
t = pd.DataFrame(result, columns=['taxi','witness_answer'])

In [166]:
t.groupby('witness_answer')['taxi'].mean()

witness_answer
0    0.042474
1    0.416890
Name: taxi, dtype: float64

Вероятность того, что зеленое такси (0) было правильно указано 0.042  
Вероятность того, что синее такси (1) было правильно указано 0.416  
Ответ: 0.416